File: Fructose.Test/Tests/PHPCall.rb
#TEST EXPECTS:
#42
#720
#from php
#from fructose

require :phpcall
require 'Tests/PHPCall.php'

puts phpcall(:foo)
puts phpcall(:fact, 6)

phpobj = phpcall :myClassFactory
phpobj.attr_set :msg, "from php"
phpobj.call :foobar
phpobj.attr_set :msg, "from fructose"
puts phpobj.attr :msg

File: Fructose.Test/Tests/PHPCall.php
<?php

function foo()
{
	return 42;
}

function fact($n)
{
	if($n <= 1)
		return 1;
	return $n * fact($n - 1);
}

class MyClass
{
    function foobar()
	{
		echo $this->msg . "\n";
	}
}
function myClassFactory()
{
	return new MyClass;
}